docs/contributing/Feature-flags.md

```diff
@@ -0,0 +1,328 @@
+# Using feature flags
+
+[flutter.dev/to/feature-flags](https://flutter.dev/to/feature-flags)
+
+The Flutter tool (`flutter`) supports the concept of _feature flags_, or boolean
+flags that can inform, change, allow, or deny access to behavior, either in the
+tool itself, or in the framework (`package:flutter`, and related).
+
+> [!WARNING]
+>
+> This document is based on the unmerged PR [#168437](https://github.com/flutter/flutter/pull/168437).
+
+---
+
+Table of Contents
+
+- [Overview](#overview)
+  - [Why feature flags](#why-feature-flags)
+- [Adding a flag](#adding-a-flag)
+  - [Allowing flags to be enabled](#allowing-flags-to-be-enabled)
+  - [Enabling a flag by default](#enabling-a-flag-by-default)
+  - [Removing a flag](#removing-a-flag)
+- [Using a flag to drive behavior](#using-a-flag-to-drive-behavior)
+  - [Tool](#tool)
+  - [Framework](#framework)
+  - [Tests](#tests)
+
+## Overview
+
+For example, [enabling the use of Swift Package Manager][enable-spm]:
+
+```sh
+flutter config --enable-swift-package-manager
+```
+
+Feature flags can be configured globally (for an entire _machine_), locally
+(for a particular _app_), per-test, and be automatically enabled for different
+release channels (`master`, versus `beta`, versus `stable`), giving multiple
+consistent options for developing.
+
+_See also, [Flutter pubspec options > Fields > Config][pubspec-config]._
+
+[enable-spm]: https://docs.flutter.dev/packages-and-plugins/swift-package-manager/for-app-developers
+[pubspec-config]: https://docs.flutter.dev/tools/pubspec#config
+
+### Why feature flags
+
+Feature flags allow conditionally, consistently, and covenentiantly change
+behavior.
+
+For example:
+
+- **Gradual rollouts** to introduce new features to a small subset of users.
+
+- **A/B Testing** to easily test or compare different implementations.
+
+- **Kill Switches** to quickly disable problematic features without large code
+  changes.
+
+- **Allow experimental access** to features not ready for broad or unguarded
+  use.
+
+  We do not consider it a breaking change to modify or remove experimental flags
```

Comment:
We do not consider changes guarded by experimental flags to be breaking either. These features are _experimental_ - we are seeking feedback.

Reply:
Done, tweaked language - PTAL.